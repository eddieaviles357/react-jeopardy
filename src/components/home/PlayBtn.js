import React from 'react'
import './home.css'

const PlayBtn = () => {
  return (
    <button className='Play-btn btn'>Play</button>
  )
}

export default PlayBtn
