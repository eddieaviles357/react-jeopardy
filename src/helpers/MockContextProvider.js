"use strict";

import { useState } from 'react'
import JeopardyContext from '../JeopardyContext'

const MockContextProvider = ({ children }) => {
    const [jpardyData, setJpardyData] = useState([])
    const [isPlaying, setIsPlaying] = useState(false)
    const [isLoading, setIsLoading] = useState(false)
    // used to avoid multiple calls to api
    const [isPlayedBtnStarted, setisPlayedBtnStarted] = useState(false)

    const startGame = async () => {
        try {
            if (isPlayedBtnStarted) return
            setisPlayedBtnStarted(true)
            setIsLoading(true)
            setJpardyData(await setBoardData())
            setIsPlaying(true)
            setIsLoading(false)
        } catch (err) {
            return <div>Sorry something went wrong.Please Refresh</div>
        }
    }
    return (
        <JeopardyContext.Provider value={{
            jpardyData,
            setJpardyData,
            isPlaying,
            setIsPlaying,
            startGame,
            isLoading,
        }}>
            {children}
        </JeopardyContext.Provider>
    )
}

export default MockContextProvider;