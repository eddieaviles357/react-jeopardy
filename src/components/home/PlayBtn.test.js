"use strict";

import { render, screen } from '@testing-library/react';
import PlayBtn from "./PlayBtn";
import MockContextProvider from "../../helpers/MockContextProvider";

test('does not crash', () => {
    render(
        <MockContextProvider>
            <PlayBtn />
        </MockContextProvider>);
});

test('matches snapshot', () => {
    const playBtn = render(
        <MockContextProvider>
            <PlayBtn />
        </MockContextProvider>);
    expect(playBtn.asFragment()).toMatchSnapshot();
})

describe('should render Loading if prop loading is true', () => {
    const playBtn = render(
        <MockContextProvider initialProps={{ loading: true }}>
            <PlayBtn />
        </MockContextProvider>
    );
    const loading = screen.getByText(/Loading/i);
    expect(loading.textContent).toBe('Loading');
})