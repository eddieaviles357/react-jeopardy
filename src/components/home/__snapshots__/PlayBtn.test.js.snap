// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="Play-btn btn"
    tabindex="0"
  >
    Play
  </button>
</DocumentFragment>
`;
